services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/team/TeamManagerImpl.java

```diff
@@ -707,6 +707,28 @@ public String getIconURL(UserInfo userInfo, String teamId) throws NotFoundExcept
 		return fileHandleManager.getRedirectURLForFileHandle(urlRequest);
 	}
 
+	@Override
+	public String getIconPreviewURL(UserInfo userInfo, String teamId) throws NotFoundException {
+		Team team = teamDAO.get(teamId);
+
+		String fileHandleId = team.getIcon();
+
+		if (fileHandleId == null) {
+			throw new NotFoundException("Team " + teamId + " has no icon file handle.");
```

Comment:
We can extract a method to get the file handle id of a team since the same logic is used in more than one place.